pkg/envoy/xds/ack.go

```diff
@@ -163,7 +169,20 @@ func (m *AckingResourceMutatorWrapper) Upsert(typeURL string, resourceName strin
 			// We don't know whether the revert did an Upsert or a Delete, so as a
 			// best effort, just wait for any ACK for the version and type URL,
 			// and ignore the ACKed resource names, like for a Delete.
-			m.addDeleteCompletion(typeURL, version, nodeIDs, completion)
+			m.addVersionCompletion(typeURL, version, nodeIDs, completion)
+		}
+	}
+}
+
+func (m *AckingResourceMutatorWrapper) UseCurrent(typeURL string, nodeIDs []string, wg *completion.WaitGroup) {
+	m.locker.Lock()
+	defer m.locker.Unlock()
+
+	if wg != nil {
```

Comment:
It seems weird to have this check here, why do we have it?

Reply:
It was there to avoid using a nil pointer, but for this code path the pointer is never nil, so I removed the check and added comments to the callers that the WaitGroup may not be nil.

On the corresponding UpdateNetworkPolicy path we explicitly set `wg` to `nil` when we know that there is no proxy running (yet) to ACK the changes.